### PR TITLE
Support min and max with datetime64

### PR DESCRIPTION
**Pull request by SimonHeybrock**

### CODE REVIEW

File: core/include/scipp/core/element/comparison.h [hunk 1]
```diff
@@ -91,44 +91,44 @@ constexpr auto equal = overloaded{
...
 
-constexpr auto max_equals =
-    overloaded{arg_list<double, float, int64_t, int32_t, bool>,
-               transform_flags::expect_in_variance_if_out_variance,
-               [](auto &&a, const auto &b) {
-                 using std::max;
-                 a = max(a, b);
-               }};
+constexpr auto max_equals = overloaded{
+    arg_list<double, float, int64_t, int32_t, bool, core::time_point>,
```
Comment (nvaytet) [inline core/include/scipp/core/element/comparison.h:95]: These changes look similar to @jl-wynen 's PR #1747
we should probably coordinate?

Comment (SimonHeybrock) [reply core/include/scipp/core/element/comparison.h:95]: Good point, I forgot...

Comment (SimonHeybrock) [reply core/include/scipp/core/element/comparison.h:95]: Switched to using @jl-wynen 's implementation of `creation.h`, which is cleaner than what I had.

**Review by jl-wynen — APPROVED** [review]
Compared with my implementation. Looks equal, so I am happy :)

File: core/include/scipp/core/element/comparison.h (hunk 1 above)
Comment (nvaytet) [reply core/include/scipp/core/element/comparison.h:95]: in his PR he seems to not need the `core::` namespace specification here..

**Pull request merged**